src/data_structures/bwt.rs

```diff
@@ -37,7 +46,7 @@ pub type BWTFind = Vec<usize>;
 /// let bwt = bwt(text, &pos);
 /// assert_eq!(bwt, b"ATTATTCAGGACCC$CTTTCAA");
 /// ```
-pub fn bwt(text: &[u8], pos: &SuffixArraySlice) -> BWT {
+pub fn bwt<SA: Deref<Target = RawSuffixArray>>(text: &[u8], pos: SA) -> BWT {
```

Comment:
I don't think these functions should take the deref. The most general way is to require a normal reference here. We should only use the deref where we need ownership and want to allow wrapper types like RC as well.